helm-charts/splunk-otel-collector/templates/daemonset.yaml

```diff
@@ -252,8 +252,14 @@ spec:
         {{- if or $agent.securityContext (and (eq (include "splunk-otel-collector.logsEnabled" $) "true") (eq .Values.logsEngine "otel")) }}
         securityContext:
           {{- if $agent.securityContext }}
-          {{- toYaml $agent.securityContext | nindent 10 }}
-          {{- else }}
+            {{- if .Values.isWindows }}
+              {{- with unset $agent.securityContext "runAsUser" }}
+              {{- toYaml $agent.securityContext | nindent 10 }}
```

Comment:
Maybe I'm missing something, but if this line should be run whether or not it's a Windows system, we can just have it once outside of the `if..else..` instead of inside both `if` and `else`. Admittedly I'm not very experienced with YAML's formatting here, so I may be misunderstanding this.

Same goes for changes in the other files.

Reply:
@crobert-1 
For windows, we are running `with unset...` , if it's windows, then we just use the non-altered  `$agent.securityContext` 
The unset context will be lost if I run it outside the `with unset` block

Reply:
Okay, thanks for clarifying!